pxd.c

```diff
@@ -1520,24 +1592,38 @@ ssize_t pxd_remove(struct fuse_conn *fc, struct pxd_remove_out *remove)
 	pr_info("removing device %llu", pxd_dev->dev_id);
 
 	/* Make sure the req_fn isn't called anymore even if the device hangs around */
-	if (pxd_dev->disk && pxd_dev->disk->queue){
+	if (pxd_dev->exported && pxd_dev->disk && pxd_dev->disk->queue){
```

Comment:
If the queue exists, it should be marked dying. Doesn't matter exported is set or not.

Reply:
exported is set to true only if disk and disk queue both are setup properly.
If during setup, either of them fail, then it is all cleaned up there, so one cannot have disk without queue.
The added check should not hurt.